core/server/worker/src/main/java/alluxio/worker/block/MonoBlockStore.java

```diff
@@ -331,7 +331,15 @@ public CompletableFuture<List<BlockStatus>> load(List<Block> blocks, UfsReadOpti
         handleException(e, block, errors, sessionId);
         continue;
       }
-      ByteBuffer buf = NioDirectBufferPool.acquire((int) blockSize);
+      ByteBuffer buf;
```

Comment:
If we want to retry getting buffer, we probably add a dedicated function to retry and handle OutOfMemoryError specifically. I personally don't want RetryUtils.retryCallable to handle Throwable in case we swallow unwanted Errors

Reply:
Also, we should tune the workerResource/loadBatchSize instead of retrying here